compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -276,12 +306,12 @@ struct FoldFillWithSetEncoding final
   }
 };
 
-struct SetEncodingPass final : impl::SetEncodingPassBase<SetEncodingPass> {
+struct SetEncodingPass : public impl::SetEncodingPassBase<SetEncodingPass> {
```

Comment:
Final is better in this case, as we do not expect any other class to inherent it.

https://en.cppreference.com/w/cpp/language/final